### DISCUSSION TITLE
How do you speed up the development process?

### DISCUSSION THREAD

**Discussion by MartinXPN** [ManimCommunity/manim#3168] [Q&A]
Hey everyone,
I'm still kinda new to Manim (only created a handful of animations) and I was wondering how you guys speed up your development process?
I realized that adding mobjects to a scene continuously makes the process slow down over time. So I started doing a trick where I would prepare the scene for a transition at the end of the `construct` method and then re-create the last frame at the beginning of the next scene and take it from there.

This makes iterations faster but results in some boilerplate code that might be avoided.

What's the proper way of handling this kind of situations? How do you add things to scenes without making the iteration process slower?

**Comment by MrDiver** [comment]
Most likely depends on the scene you're creating. But avoiding copying is a good start. So reuse as many objects as possible to avoid the garbage collector running into the animation every second because 100 copies need to be deleted. (Might be a bit unintuitive which things cause copies, but we currently have a lot of them for Animations).

For example Updaters are really the Kings of deepcopy in that case. So if you want to go fast just avoid them as long as it works without them. Otherwise try to combine Simple things into animation groups to reduce the number of play calls.
